remme/atomic_swap_tp/client.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2018 REMME
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import json
+import datetime
+
+import logging
+
+from remme.atomic_swap_tp.handler import AtomicSwapHandler
+from remme.protos.atomic_swap_pb2 import AtomicSwapInitPayload, AtomicSwapExpirePayload, AtomicSwapClosePayload, \
+    AtomicSwapMethod, AtomicSwapInfo, AtomicSwapSetSecretLockPayload, AtomicSwapApprovePayload
+from remme.protos.certificate_pb2 import NewCertificatePayload, CertificateMethod
+from remme.settings import GENESIS_ADDRESS, SETTINGS_SWAP_COMMISSION, ZERO_ADDRESS
+from remme.settings_tp.handler import _make_settings_key
+from remme.shared.basic_client import BasicClient
+from remme.certificate_tp.handler import CertificateHandler
+from remme.shared.utils import attr_dict
+from remme.token_tp.handler import TokenHandler
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_swap_init_payload(args):
+    payload = AtomicSwapInitPayload()
+    LOGGER.info(args)
+    payload.receiver_address = args['receiver_address']
+    payload.sender_address_non_local = args['sender_address_non_local']
+    payload.amount = args['amount']
+    payload.swap_id = args['swap_id']
+    payload.secret_lock_optional_bob = args.get('secret_lock_optional_bob', "")
+    payload.email_address_encrypted_optional_alice = args.get('email_address_encrypted_optional_alice', "")
+    payload.created_at = args['created_at']
+
+    return payload
+
+
+def get_swap_approve_payload(args):
+    payload = AtomicSwapApprovePayload()
+    payload.swap_id = args['swap_id']
+
+    return payload
+
+
+def get_swap_expire_payload(args):
+    payload = AtomicSwapExpirePayload()
+    payload.swap_id = args['swap_id']
+
+    return payload
+
+
+def get_swap_set_secret_lock_payload(args):
+    payload = AtomicSwapSetSecretLockPayload()
```

Comment:
Is there a reason to pass **args**, but not named parameters ?

Reply:
yes, minimize code duplication, previously added attr_dict to get rid of dict syntax